csrc/host_ir/jit.cpp

```diff
@@ -166,6 +178,401 @@ void printLlvmIr(llvm::Function* func, std::string_view msg) {
+  } else if(binary_op->getBinaryOpType() == BinaryOpType::Sub) {
+    out_value = builder.CreateSub(lhs_value, rhs_value);
+  } else if(binary_op->getBinaryOpType() == BinaryOpType::Mul) {
+    out_value = builder.CreateMul(lhs_value, rhs_value);
+  } else if(binary_op->getBinaryOpType() == BinaryOpType::CeilDiv) {
+    // Implement ceilDiv as (a + b - 1) / b
+    llvm::Value* numerator =
+    builder.CreateAdd(lhs_value, rhs_value);
+    llvm::Value* one = builder.getInt64(1);
+    numerator = builder.CreateSub(numerator, one);
+    out_value = builder.CreateUDiv(numerator, rhs_value);
+  } else {
+    NVF_THROW("LLVM Lowering Error: Unsupported binary operation type in extent calculation: ", binary_op->getBinaryOpType());
+  }
+  return out_value;
+}
+
+// Helper function to create value for unary operation
+llvm::Value* createValueForUnaryOp(UnaryOp* unary_op, std::unordered_map<Val*, llvm::Value*>& val_to_value, llvm::IRBuilder<>& builder) {
+  auto* in = unary_op->in()->as<Val>();
+  llvm::Value* in_value = getOrCreateValueForExtent(in, val_to_value, builder);
+  llvm::Value* out_value = nullptr;
+  if(unary_op->getUnaryOpType() == UnaryOpType::Cast) {
+    out_value = in_value;
+  } else if(unary_op->getUnaryOpType() == UnaryOpType::Abs) {
+    llvm::Value* is_negative = builder.CreateICmpSLT(in_value, builder.getInt64(0));
+    llvm::Value* negated = builder.CreateNeg(in_value);
+    out_value = builder.CreateSelect(is_negative, negated, in_value);
+  } else if(unary_op->getUnaryOpType() == UnaryOpType::Neg) {
+    out_value = builder.CreateNeg(in_value);
+  } else {
+    NVF_THROW("LLVM Lowering Error: Unsupported unary operation type in extent calculation: ", unary_op->getUnaryOpType());
+  }
+  return out_value;
+}
+// Helper function to create value for extent
+llvm::Value* createValueForExtent(
+    Val* val,
+    std::unordered_map<Val*, llvm::Value*>& val_to_value,
+    llvm::IRBuilder<>& builder) {
+  llvm::Value* out_value = nullptr;
+  if (val->isA<IterDomain>()) {
+    if(val->as<IterDomain>()->isBroadcast()) {
+      out_value = builder.getInt64(1);
+      if (val->as<IterDomain>()->hasExpandedExtent()) {
+        out_value = getOrCreateValueForExtent(val->as<IterDomain>()->expandedExtent(), val_to_value, builder);
+      } 
+    } else {
+      out_value = getOrCreateValueForExtent(val->as<IterDomain>()->extent(), val_to_value, builder);
+    }
+  } else if (val->isConst()) {
+    out_value = builder.getInt64(val->value().as<int64_t>());
+  } else if (Expr* def = val->definition()) {
+    if (auto* binary_op = def->as<BinaryOp>()) {
+      out_value = createValueForBinaryOp(binary_op, val_to_value, builder);
+    } else if (auto* unary_op = def->as<UnaryOp>()) {
+      out_value = createValueForUnaryOp(unary_op, val_to_value, builder);
+    } else {
+      NVF_THROW(
+        "LLVM Lowering Error: createValueForExtent called with unsupported "
+        "operation type: ",
+        def->getOpString());
+    }
```

Comment:
these are fine as-is. The size of this will explode as we try to support more fusion ops.